components/EmotionDetails.js

```diff
@@ -2,19 +2,57 @@ import EmotionCard from "./EmotionCard";
 import Heading from "./Heading";
 import styled from "styled-components";
 import Link from "next/link";
+import { useState } from "react";
+
+export default function EmotionDetails({ object, onDeleteEmotion }) {
+  const [isDialogOpen, setIsDialogOpen] = useState(false); // State to manage delete-dialog visibility
+
+  // Function to open the confirmation dialog
+  function handleOpenDialog() {
+    setIsDialogOpen(true);
+  }
+
+  // Function to close the confirmation dialog
+  function handleCloseDialog() {
+    setIsDialogOpen(false);
+  }
+
```

Comment:
Instead of using two functions, you could also write  one toggle function: `function toggleDeleteDialog() {
    setIsDialogOpen(!isDialogOpen);
  }`